File: B_Worms.py
def ans(a,x):
    l=0
    r=len(a)-1
    while l<=r:
        mid=(l+r)//2
        if a[mid]==x: return mid
        elif a[mid]<=x: l=mid+1
        else: r=mid-1
    return l
n=int(input())
a=list(map(int,input().split()))
prefix=[]
for i in range(n):
    if i: prefix.append(prefix[i-1]+a[i])
    else: prefix.append(a[i])
m=int(input())
q=list(map(int,input().split()))
# print(prefix)
for it in q:
    res=ans(prefix,it)
    print(res+1)
# 2 9 12 16 25 --1
# 0 4 2 12
# 0 1 0 2
# 1 1
# 
# 

File: B_Worms.cpp
#include<bits/stdc++.h>
#include<ext/pb_ds/assoc_container.hpp>
#include<ext/pb_ds/tree_policy.hpp>
using namespace std;
using namespace __gnu_pbds;
#define Oset tree<int, null_type, less<int>, rb_tree_tag, tree_order_statistics_node_update>
#define fast_IO ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);
#define w(t) int t; cin>>t; while(t--)
#define nn "\n"
#define int long long
#define pb push_back
#define ld long double
int binPow(int n, int k){
    if(k==0) return 1;
    else{
        int turj=1;
        while(k){if(k&1) turj*=n; n*=n, k>>=1;}
        return turj;
    }
}
bool compare(pair<int,int>&a, pair<int,int>&b){
    if(a.first!=b.first){
        if(a.first<b.first) return false;
        return true;
    }else{
        if(a.second>b.second) return false;
        return true;
    }
}
int pile(vector<int>&v,int x){
    int n=v.size(), l=0, r=n-1, ans=n;
    while(l<=r){
        int mid=(l+r)/2;
        if(v[mid]>=x) r=mid-1;
        else l=mid+1;
    }
    return l;
}
int32_t main(){
    fast_IO
    int n; cin>>n;
    vector<int>v(n),prefix(n+1,0);
    for(int i=0; i<n; i++){
        cin>>v[i];
        prefix[i+1]=prefix[i]+v[i];
    }
    // for(auto &it: prefix) cout<<it<<" "; cout<<nn;
    w(q){
        int b; cin>>b;
        cout<<pile(prefix,b)<<nn;
    }
    return 0;
}